phys2bids/tests/test_physio_obj.py

```diff
@@ -89,7 +91,7 @@ def test_BlueprintInput():
     test_time = np.array([0, 1, 1, 2, 3, 5, 8, 13])
     test_trigger = np.array([0, 1, 0, 0, 0, 0, 0, 0])
     test_chocolate = np.array([1, 0, 0, 1, 0, 0, 1, 0])
-    test_timeseries = [test_time, test_trigger, test_chocolate]
+    test_timeseries = [deepcopy(test_time), test_trigger, test_chocolate]
```

Comment:
Why deepcopy this array?

Reply:
If we don't make a deep copy, it's just a shallow reference, so `test_time` changes when `blueprint_in.check_trigger_amount()` is called, and we wouldn't be able to set up the new asserts.

Reply:
I think that's a bug on `BlueprintInput`'s end rather than something that the test should work around.

Reply:
Actually this is related to the effect of shallow copies vs deepcopies: Basically, a shallow copy does copy the main object but references child objects to the original. Therefore if you modify the child object of the original, it also gets modified in the copy. If you deepcopy, you create a completely independent object (that consumes more memory). More info:
https://realpython.com/copying-python-objects/

Reply:
It is an undesired behavior but not a bug, just how python works

Reply:
Yup, @vinferrer is right. For our scopes, I don't think it's an undesired behaviour either.
However, if @tsalo you think there's a better solution to approach this, we can open an issue to change this - I wouldn't block this PR for this, though.

Reply:
When I say it is a bug, I mean that a given function/class should not to modify its inputs directly unless that behavior is desired and documented. A good example of this is [`nilearn.image.get_data()`](https://nilearn.github.io/modules/generated/nilearn.image.get_data.html#nilearn.image.get_data), which is an alternative to nibabel's Nifti1Image's `get_fdata()`. Unlike nibabel's method, nilearn returns an img object's numpy array directly instead of a copy. Even with that warning, there are still unexpected behaviors like https://github.com/nilearn/nilearn/issues/2548, so I'd lean against it. That issue I linked was only a bug _after_ they switched from nibabel's function (return a copy) to nilearn's function (return a reference).

On the other hand, when the behavior isn't documented, we can get issues like https://github.com/nilearn/nilearn/issues/2281 and https://github.com/nilearn/nilearn/issues/2475. 

There are valid reasons to not make copies of all of your inputs within a function of course, like memory. I can't speak to the exact memory requirements of a BlueprintInput, but I imagine they could be quite large given the right circumstances. If you don't want to make a copy _within_ the BlueprintInput, the we should probably add something to the docstring about this behavior. `nilearn.image.get_data()`'s docstring could serve as a good template.

This is a minor thing, but since this numpy array is just an array of integers or floats, and not objects, shallow copying should be fine. You could use `test_time.copy()` without issue. The problem is that `BlueprintInput` doesn't make a copy of the array at all and operates on the original array directly.

Reply:
@tsalo now it's clear. Could you open an issue for this? I'm going to change the deepcopy to copy.

Reply:
I've opened #314 (what luck!). I'll resolve this thread, but fixing the bug will involve dropping the copy in the test in a later PR. Thanks!